gen/src/generate/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod char_property;
+
+mod idna;
+mod ucd;
+
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+
+/// The standard auto-generated source disclaimer.
+const PREAMBLE: &'static str = "// WARNING: Auto-generated by unic-gen. DO NOT EDIT MANUALLY!";
+
+pub fn generate_all(mut crates: Vec<String>) -> io::Result<()> {
+    crates.sort();
+
+    for krate in &crates {
+        let path = tables_path(krate);
+        if path.exists() {
+            fs::remove_dir_all(&path)?;
+        }
+        fs::create_dir_all(path)?;
+    }
+
+    idna::generate_all(crates.iter().filter(|x| x.starts_with("unic-idna")))?;
```

Comment:
Currently, it does do this (or a similar thing) currently.

`cargo run --package=unic-gen -- --crate unic-idna` will only generate tables for `unic-idna` and subcrates. `cargo run --package=unic-gen -- --crate unic-ucd` will only generate tables for `unic-ucd` and subcrates. `cargo run --package=unic-gen -- --unicode-version 10.0.0` will only download files as specified in the config.

Once we've moved all of the crates onto the new tables, though, I don't think this level of granularity, and just `ucd`, `idna`, `emoji`, or whatever should be fine enough. Currently it defaults to nothing done, and you `--crate unic` to specify to generate all tables; I personally would like to keep that behavior such that just running the binary will print help and it's simple to just download data without generating tables if you so choose.